modules/custom/az_core/src/Plugin/ConfigProvider/QuickstartConfigProvider.php

```diff
@@ -27,7 +27,111 @@ class QuickstartConfigProvider extends ConfigProviderBase {
    * {@inheritdoc}
    */
   public function addConfigToCreate(array &$config_to_create, StorageInterface $storage, $collection, $prefix = '', array $profile_storages = []) {
-    // The caller will aready have loaded config for install.
+    // Remove permissions that don't exist.
+    $config_to_create = $this->trimPermissions($config_to_create);
+  }
+
+  /**
+   * Find newly available permissions from profile.
+   *
+   * @param \Drupal\Core\Extension\Extension[] $extensions
+   *   An associative array of Extension objects, keyed by extension name.
+   *
+   * @return array
+   *   A list of the configuration data keyed by configuration object name.
+   */
+  public function findProfilePermissions(array $extensions = []) {
+
+    // Get active configuration to look for users.
+    $existing_config = $this->getActiveStorages()->listAll();
+    // phpcs:ignore
+    $existing_users = array_filter($existing_config, function ($name) {
+      return (strpos($name, 'user.role.') === 0);
+    });
+    $user_config = $this->getActiveStorages()->readMultiple($existing_users);
```

Comment:
Nitpick: roles, not users
```suggestion
    $existing_roles = array_filter($existing_config, function ($name) {
      return (strpos($name, 'user.role.') === 0);
    });
    $role_config = $this->getActiveStorages()->readMultiple($existing_roles);
```